doc/api/api_changes/2016-01-03_colorbar.rst

```diff
@@ -0,0 +1,12 @@
+`ColorBar` raises TypeError rather than ignore user input
+`````````````````````````````````````````````````````````
+
+In ``__init__`` `ColorBar` ignores user supplied ``cmap`` and ``norm``
+keyword arguments in favor of retrieving those values from the
+``mappable`` passed in.  If those keywords are supplied `ColorBar`
+will now raise a `TypeError` rather than silently drop user supplied
```

Comment:
The logic of `TypeError` is to match what would happen if you passed the incorrect number of args etc to a function.